src/components/PostDetails.tsx

```diff
@@ -1,117 +1,218 @@
-import React from 'react';
+/* eslint-disable react/jsx-one-expression-per-line */
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { Comment, CommentData } from '../types/Comment';
+import { Errors } from '../types/Errors';
+
+import { getComments, postComment, deleteComment } from '../api/users';
+
+type Props = {
+  post: Post | null;
+  user: User | null;
+  isFormShown: boolean;
+  typeOfError: Errors;
+  setTypeOfError: (data: Errors) => void;
+  showForm: () => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  user,
+  isFormShown,
+  showForm,
+  typeOfError,
+  setTypeOfError,
+}) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | null>(null);
+  const [areCommentsLoading, setCommentsLoading] = useState(false);
+  const [isCommentAdding, setCommentAdding] = useState(false);
+  const [
+    removingCommentsId, setRemovingCommentsId,
+  ] = useState<number | null>(null);
+  const [newCommentsData, setNewCommentsData] = useState<CommentData>(
+    {
+      postId: post?.id || null,
+      name: '',
+      email: '',
+      body: '',
+    },
+  );
+
+  async function loadComments() {
+    setTypeOfError(Errors.None);
+    try {
+      setCommentsLoading(true);
+      const commentsFromServer = await getComments(user?.id);
```

Comment:
Replace 'user?.id' with 'post?.id' as the getComments function should receive the postId, not the userId.